src/plugins/intel_npu/src/CMakeLists.txt

```diff
@@ -9,13 +9,9 @@ add_subdirectory(utils)
 
 add_subdirectory(al)
 
-if(ENABLE_DRIVER_COMPILER_ADAPTER AND ENABLE_ZEROAPI_BACKEND)
-    add_subdirectory(compiler)
-endif()
+add_subdirectory(compiler_adapter)
```

Comment:
When build driver compiler, we do not want to contain modules that are not required. Seems no way to remove adapter , backend from the dll now.

Reply:
yep, no way right now.